wallet-new/src/Cardano/Wallet/Kernel/Types.hs

```diff
@@ -24,12 +28,46 @@ import           Data.Word (Word32)
 import           Pos.Core (MainBlock, Tx, TxAux (..), TxIn (..), TxOut, TxOutAux (..), gbBody,
                            mbTxs, mbWitnesses, txInputs, txOutputs)
 import           Pos.Crypto.Hashing (hash)
+import           Pos.Crypto (EncryptedSecretKey)
 import           Pos.Txp (Utxo)
 import           Serokell.Util (enumerate)
 
+import qualified Cardano.Wallet.Kernel.DB.HdWallet as HD
 import           Cardano.Wallet.Kernel.DB.InDb
 import           Cardano.Wallet.Kernel.DB.Resolved
 
+{-------------------------------------------------------------------------------
+  Abstract WalletId and AccountId
+-------------------------------------------------------------------------------}
+
+-- | Wallet Id
+--
+-- A Wallet Id can take several forms, the simplest of which is a hash
+-- of the Wallet public key
+data WalletId =
+    -- | HD wallet with randomly generated addresses
+  WalletIdHdRnd HD.HdRootId
+
+    {- potential future kinds of wallet IDs:
+    -- | HD wallet with sequentially generated addresses
+    | WalletIdHdSeq ...
+
+    -- | External wallet (all crypto done off-site, like hardware wallets)
+    | WalletIdExt ...
+    -}
+
+    deriving (Eq, Ord)
+
+-- | Map of Wallet Master keys indexed by WalletId
+--
+-- TODO: We may need to rethink having this in-memory
+-- ESK should _not_ end up in the wallet's acid-state log
+type WalletESKs = Map WalletId EncryptedSecretKey
```

Comment:
Is this TODO still a concern? IIRC @edsko introduced the concept of `InDB` exactly to prevent this kind of situation from happening. Can we accidentally store an `EncryptedSecretKey` inside `acid-state`, as it stand?